src/NServiceBus.Core/obsoletes-v8.cs

```diff
@@ -606,4 +606,204 @@ public static DateTime ToUtcDateTime(string wireFormattedString)
+    public static class ConfigurationTimeoutExtensions
+    {
+        [ObsoleteEx(
+            Message = "The timeout manager has been removed. See the upgrade guide for more details",
+            TreatAsErrorFromVersion = "8",
+            RemoveInVersion = "9")]
+        public static void TimeToWaitBeforeTriggeringCriticalErrorOnTimeoutOutages(this EndpointConfiguration config, TimeSpan timeToWait)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+}
+
+namespace NServiceBus.Features
+{
+    [ObsoleteEx(
+        Message = "The timeout manager has been removed. See the upgrade guide for more details.",
+        TreatAsErrorFromVersion = "8",
+        RemoveInVersion = "9")]
+    public class TimeoutManager
+    {
+        internal TimeoutManager() => throw new NotImplementedException();
+    }
+}
+
+namespace NServiceBus.Timeout.Core
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Extensibility;
+
+    [ObsoleteEx(
+        Message = "The timeout manager has been removed. See the upgrade guide for more details.",
+        TreatAsErrorFromVersion = "8",
+        RemoveInVersion = "9")]
+    public interface IPersistTimeouts
+    {
+        Task Add(TimeoutData timeout, ContextBag context);
+
+        Task<bool> TryRemove(string timeoutId, ContextBag context);
+
+        Task<TimeoutData> Peek(string timeoutId, ContextBag context);
+
+        Task RemoveTimeoutBy(Guid sagaId, ContextBag context);
+    }
+
+    [ObsoleteEx(
+        Message = "The timeout manager has been removed. See the upgrade guide for more details.",
+        TreatAsErrorFromVersion = "8",
+        RemoveInVersion = "9")]
+    public interface IQueryTimeouts
+    {
+        Task<TimeoutsChunk> GetNextChunk(DateTimeOffset startSlice);
+    }
+
+    [ObsoleteEx(
+        Message = "The timeout manager has been removed. See the upgrade guide for more details.",
+        TreatAsErrorFromVersion = "8",
+        RemoveInVersion = "9")]
+    public class TimeoutData
+    {
+        public string Id => throw new NotImplementedException();
+
+        public string Destination => throw new NotImplementedException();
+
+        public Guid SagaId => throw new NotImplementedException();
+
+        public byte[] State => throw new NotImplementedException();
+
+        public DateTimeOffset Time => throw new NotImplementedException();
+
+        public string OwningTimeoutManager => throw new NotImplementedException();
+
+        public Dictionary<string, string> Headers => throw new NotImplementedException();
+    }
+
+    [ObsoleteEx(
+        Message = "The timeout manager has been removed. See the upgrade guide for more details.",
+        TreatAsErrorFromVersion = "8",
+        RemoveInVersion = "9")]
+    public class TimeoutsChunk
+    {
+        public TimeoutsChunk(Timeout[] dueTimeouts, DateTimeOffset nextTimeToQuery)
+        {
+            throw new NotImplementedException();
+        }
```

Comment:
This is a new api, must be the old
```suggestion
        public TimeoutsChunk(Timeout[] dueTimeouts, DateTime nextTimeToQuery)
        {
            throw new NotImplementedException();
        }
```